prune.go

```diff
@@ -341,13 +344,15 @@ func PruneTxns(db *mgo.Database, oracle Oracle, txns *mgo.Collection, stats *Cle
 		}
 	}
 	if len(toRemove) > 0 {
-		removedCount += len(toRemove)
-		if err := oracle.RemoveTxns(toRemove); err != nil {
+		referencedCount += len(toRemove)
+		if count, err := oracle.RemoveTxns(toRemove); err != nil {
 			return fmt.Errorf("removing completed txns: %v", err)
+		} else {
+			removedCount += count
 		}
 		toRemove = toRemove[:0]
 	}
-	logger.Debugf("%d txns are still referenced and will be kept", removedCount)
+	logger.Debugf("%d txns are still referenced and will be kept (%d we would have removed)", referencedCount, removedCount)
```

Comment:
I went with:
	// We don't expect 'removedCount' to be nonzero because all of them
	// should have been handled by the Clean pass.
	logger.Debugf("%d txns are still referenced and will be kept (%d unexpected references)",
		referencedCount, removedCount)